src/test/java/io/confluent/examples/streams/CustomJoinWithTableTriggeringStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,479 @@
+ * 20   |                     ("alice", 1L) | -
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ * Kafka Streams LEFT stream-table join:
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | ("alice", (999.99, null)
+ * 20   |                     ("alice", 1L) | -
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ *
+ * The code in this example changes the above behavior so that an application will wait a configurable amount of time
+ * for data to arrive also at the table before it produces the join output for a given key (here: "alice").  The
+ * motivation is that, in this example, we'd prefer to receive fully populated join messages rather than join messages
+ * where the table-side information is missing (null).  Depending on your use case, you might prefer this changed
+ * behavior, and these changed semantics, over the default join semantics of Kafka Streams.
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | -
+ * 20   |                     ("alice", 1L) | ("alice", (999.99, 1L))
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ * Note how, in the example above, the custom join produced a join output of `("alice", (999.99, 1L))`, even though the
+ * table-side record `("alice", 1L)` arrived AFTER the stream record `("alice", 999.99)`.
+ *
+ * IMPORTANT: Kafka Streams' current stream-table join semantics dictate that only a single join output will ever be
+ * produced for a newly arriving stream-side record.  Table-side triggering of the join should only be used to ensure
+ * (rather: to increase the chance) that, when a join output is actually produced, it contains data from both the stream
+ * and the table.  However, table-side triggering should NOT be used to sent multiple join outputs for the same
+ * stream-side record.
+ *
+ * Example of WRONG table-side join triggering:
+ *
+ *           Time | Stream              Table         | Join output
+ *           -----+-----------------------------------+------------------------
+ *           10   | ("alice", 999.99)                 |
+ *           20   |                     ("alice", 1L) | ("alice", (999.99, 1L))
+ *           30   |                     ("alice", 2L) | ("alice", (999.99, 2L))
+ *
+ * In the wrong example above, only one join output must be produced, not two.  It's up to you to decide which one,
+ * however.
+ *
+ *
+ * HOW TO ADAPT THIS EXAMPLE TO YOUR OWN USE CASES
+ * ===============================================
+ *
+ * 1. You might want to add further logic that, for instance, changes the join behavior depending on the respective
+ * timestamps of received stream records and table records.
```

Comment:
This says, that the example does not use event-time semantics (what contradicts your comments that this example does use event-time semantics. I am not saying that we need to add an event-time semantics example, only, that we need to document it clearly.

I think the missing piece in our discussion is the question: why do you want table side triggering? For event-time semantics, table-time triggering only makes sense if you assume to have out-of-order data. If there is no out-of-order data, a table record would never be late, because since 2.1 we have proper timestamp synchronization (plus partial blocking if one input side does not deliver data for some time).

Handling out-of-order data correctly, is not resolved with this example fully either... Thus, I think the example does only make sense processing-time based.